src/crystal/system/print_error.cr

```diff
@@ -14,6 +14,23 @@ module Crystal::System
     {% end %}
   end
 
+  {% if flag?(:win32) %}
+    # Helper to print wide char slices (UTF-16) from the Win32 API as multibyte (UTF-8).
+    def self.print_error(bytes : Slice(UInt16)) : Nil
+      utf8 = uninitialized UInt8[256]
+      len = LibC.WideCharToMultiByte(LibC::CP_UTF8, 0, bytes, bytes.size, utf8, utf8.size, nil, nil)
```

Comment:
**issue:** We should add error handling for this function call.
It can easily fail when the slice is too big to fit into the buffer. Might not be very likely, but it can happen.  The Win32 API may return data of any size and we have no control over that. Nor can we calculate how much space a given wide char string requires in UTF-8 and fits into the buffer.

In case the buffer size is insufficient, we could process in batches of size `2 * utf8.size // 3` (`- 1` in case the last code unit is a high surrogate).

Reply:
Better: we can do it in pure Crystal with `String.each_utf16_char(slice, &)`.